client/src/components/AuthProvider.ts

```diff
@@ -67,10 +69,11 @@ export class SASAuthProvider implements AuthenticationProvider, Disposable {
 
   async createSession(): Promise<AuthenticationSession> {
     const activeProfile = profileConfig.getActiveProfileDetail();
+    const profile = <ViyaProfile>activeProfile.profile;
```

Comment:
In my opinion, we should try best to avoid type casting, to better utilize Typescript's type checking.
We can define like
```typescript
interface RestProfile {
  type: 'rest';
  myrestoption: string;
}
interface SshProfile {
  type: 'ssh';
  mysshoption: string;
}
type Profile = RestProfile | SshProfile;
```
then Typescript can automatically narrow down the type for you.
```typescript
function test(a: Profile) {
  if (a.type === 'rest') console.log(a.myrestoption); // ok
  if (a.type === 'rest') console.log(a.mysshoption); // error
}
```

Reply:
Thanks @scnwwu. Should we consider enforcing this with an eslint rule? For example, we could set [this](https://typescript-eslint.io/rules/consistent-type-assertions/) rule option to never?

Reply:
Sounds good to me.